src/figuresClasses.ts

```diff
@@ -1,19 +1,89 @@
-export interface Figure {
+const ERROR_ANY_SIDE_BELOW_OR_IS_ZERO = 'Error, any side length is <= 0';
+const ERROR_CANNOT_FORM_TRIANGLE
+  = 'Error, the longest side of a triangle is >= than a sum of two others';
 
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
 }
 
-export class Triangle {
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  public shape :Shape = Shape.Triangle;
```

Comment:
Please fix the formatting:

```suggestion
  public shape: Shape = Shape.Triangle;
```